From d99bfffa5c84435741327f544b122eb9ef5799cd Mon Sep 17 00:00:00 2001
From: Steve Lhomme <robux4@videolabs.io>
Date: Tue, 4 Jul 2017 15:24:31 +0200
Subject: [PATCH 2/2] decoder: don't create a new vout when just the multiview
 mode changes

Also don't call DecoderUpdateFormatLocked() again if it was called just before.

If the vout doesn't handle the multiview mode, recreating it won't change anything.
---
 src/input/decoder.c | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/src/input/decoder.c b/src/input/decoder.c
index dac3137..a8500fd 100644
--- a/src/input/decoder.c
+++ b/src/input/decoder.c
@@ -418,8 +418,7 @@ static int vout_update_format( decoder_t *p_dec )
      || p_dec->fmt_out.i_codec != p_owner->fmt.video.i_chroma
      || (int64_t)p_dec->fmt_out.video.i_sar_num * p_owner->fmt.video.i_sar_den !=
         (int64_t)p_dec->fmt_out.video.i_sar_den * p_owner->fmt.video.i_sar_num ||
-        p_dec->fmt_out.video.orientation != p_owner->fmt.video.orientation ||
-        p_dec->fmt_out.video.multiview_mode != p_owner->fmt.video.multiview_mode )
+        p_dec->fmt_out.video.orientation != p_owner->fmt.video.orientation )
     {
         vout_thread_t *p_vout;
 
@@ -533,14 +532,16 @@ static int vout_update_format( decoder_t *p_dec )
             return -1;
         }
     }
-
+    else
     if ( memcmp( &p_dec->fmt_out.video.mastering,
                  &p_owner->fmt.video.mastering,
                  sizeof(p_owner->fmt.video.mastering)) ||
          p_dec->fmt_out.video.lighting.MaxCLL !=
          p_owner->fmt.video.lighting.MaxCLL ||
          p_dec->fmt_out.video.lighting.MaxFALL !=
-         p_owner->fmt.video.lighting.MaxFALL)
+         p_owner->fmt.video.lighting.MaxFALL ||
+         p_dec->fmt_out.video.multiview_mode !=
+         p_owner->fmt.video.multiview_mode )
     {
         /* the format has changed but we don't need a new vout */
         vlc_mutex_lock( &p_owner->lock );
-- 
2.10.1.windows.1

